contribute/design/1.x/migrate/spc-to-cspc-migration.md

```diff
@@ -0,0 +1,205 @@
+---
+oep-number: draft Migrate 20191115
+title: Migration of SPC to CSPC
+authors:
+  - "@shubham14bajpai"
+owners:
+  - "@amitkumardas"
+  - "@vishnuitta"
+  - "@kmova"
+editor: "@shubham14bajpai"
+creation-date: 2019-11-15
+last-updated: 2020-05-21
+status: implementable
+see-also:
+  - NA
+replaces:
+  - current SPC with CSPC
+superseded-by:
+  - NA
+---
+
+# Migrate SPC to CSPC
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+* [Proposal](#proposal)
+    * [Proposed Implementation](#proposed-implementation)
+    * [High Level Design](#high-level-design)
+* [Infrastructure Needed](#infrastructure-needed)
+
+## Summary
+
+This design is aimed at providing a design for migrating SPC to CSPC
+via kubernetes job which will take SPC-name as input.
+
+This proposed design will be rolled out in phases. At a high level the design is
+implemented in the following phases:
+- Phase 1: Ability to perform migration from SPC to CSPC using
+  a Kubernetes Job.
+- Phase 2: Allow for migration of old CStor volumes to CSI CStor volumes.
+
+## Motivation
+
+### Goals
+
+- Ease the process of migrating SPC to CSPC by automating it via kubernetes job.
+- Enable day 2 operations on CStor Volumes by migrating them to CSI volumes.
+
+## Proposal
+
+### Proposed Implementation
+
+This design proposes the following key changes while migrating from a SPC to CSPC:
+
+  1. The SPC label and finalizer on BDCs will be replaced by CSPC label and finalizer. This is done to avoid webhook validation failure while creating equivalent CSPC.
+
+  2. Equivalent CSPC CR will be created for the migrating SPC.
```

Comment:
Hope tolerations, priorityclass, resource limits present on the pool deployment will be converted into corresponding pool specs and then CSPC will be created.

Reply:
yes